gammapy/spectrum/flux_point.py

```diff
@@ -670,3 +670,167 @@ def plot(self, ax=None):
             ax = plt.gca()
 
         # TODO
+
+
+def chi2_flux_points(flux_points_list, models):
+    """
+    Chi2 statistics for a list of flux points and model.
+    """
+    # extract first model from list
+    gp_model = models[0]
+    stats, stat_per_bins = 0, []
+    for flux_points in flux_points_list:
+        model = gp_model(flux_points.e_ref)
+        data = flux_points.table['dnde'].quantity
+        data_err = flux_points.table['dnde_err'].quantity
+        stat_per_bin = ((data - model) / data_err).value ** 2
+        stats += np.nansum(stat_per_bin)
+        stat_per_bins = np.append(stat_per_bins, stat_per_bin)
+    return stats, stat_per_bins
+
+
+class FluxPointsFitter(object):
+    """
+    Fit a set of flux points with a parametric model.
+
+    Parameters
+    ----------
+    optimizer : ['simplex', 'moncar', 'gridsearch']
+        Which optmizer to use.
+
+
+    Examples
+    --------
+
+    from astropy import units as u
+    from gammapy.spectrum import FluxPoints, FluxPointsFitter
+    from gammapy.spectrum.models import Powerlaw
+
+    filename = '$GAMMAPY_EXTRA/test_datasets/spectrum/flux_points/flux_points.fits'
+    flux_points = FluxPoints.read(filename)
+    fitter = FluxPointsFitter([flux_points])
```

Comment:
I don't think this example works any more. API changed.
Suggest to remove it and to point to the high-level docs for an example.